corehq/apps/notifications/models.py

```diff
@@ -58,7 +61,35 @@ def _fmt_note(note_idx):
             }
             return note_dict
 
-        return list(map(_fmt_note, enumerate(notes)))
+        notifications = list(map(_fmt_note, enumerate(notes)))
+
+        # pushes feature discovery notification to new web users
+        if not notifications:
+            all_notes = cls.objects.filter(Q(domain_specific=False) | Q(domains__overlap=couch_user.domains),
+                                        is_active=True)[:limit]
+            notifications = list(map(_fmt_note, enumerate(all_notes)))
+            notifications = [notif for notif in notifications
+                             if notif['type'] == 'feat_basic' or notif['type'] == 'feat_pro']
+
+        def remove_feature_notifications(feat_basic=False, feat_pro=False, both=False):
+            if both:
+                return [notif for notif in notifications
+                        if notif['type'] != 'feat_basic' and notif['type'] != 'feat_pro']
+            if feat_basic:
+                notif = [notif for notif in notifications if notif['type'] != 'feat_basic']
+            if feat_pro:
+                notif = [notif for notif in notifications if notif['type'] != 'feat_pro']
+            return notif
+
+        if is_USH_or_Solutions:
+            return remove_feature_notifications(both=True)
+        else:
+            if any(p in str(plan) for p in ['Pro', 'Advanced', 'Enterprise']):
+                if has_cs_groups:
+                    return remove_feature_notifications(both=True)
+                return remove_feature_notifications(feat_basic=True)
+            else:
+                return remove_feature_notifications(feat_pro=True)
```

Comment:
Since there's only going to be a max of 2 feature discovery notifications that could be removed from the initial list, fetching 12 would be enough. 

However, this code won't really be used as it currently is for anything other than to highlight case sharing groups, so there's very little chance any user is gonna see more than one feature discovery notification. If we do end up needing to generalize feature highlighting this is something that would definitely need to integrated though.